src/Symfony/Bundle/FrameworkBundle/Test/HttpClientAssertionsTrait.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Test;
+
+/*
+ * @author Mathieu Santostefano <msantostefano@protonmail.com>
+ */
+
+use Symfony\Bundle\FrameworkBundle\KernelBrowser;
+use Symfony\Component\HttpClient\DataCollector\HttpClientDataCollector;
+
+trait HttpClientAssertionsTrait
+{
+    public static function assertHttpClientHasBeenCalled(KernelBrowser $client, string $httpClientId, array $urls): void
+    {
+        if (($profile = $client->getProfile()) === false) {
+            static::fail('The Profiler must be enabled for the current request. Please ensure you have called $client->enableProfiler() before making the request.');
+        }
+
+        /** @var HttpClientDataCollector $httpClientDataCollector */
+        $httpClientDataCollector = $profile->getCollector('http_client');
+
+        foreach ($urls as $i => $url) {
+            self::assertSame($url, $httpClientDataCollector->getClients()[$httpClientId]['traces'][$i]['info']['url']);
```

Comment:
should we enforce that urls in the tests and registered traces are in the same order ?

Reply:
I think it's useful to ensure URLs are called in a certain order. Do you think we should do it in an another way?

Reply:
Well, I'm wondering whether we should _enforce_ that part of the check (after all, the client itself might run requests concurrently depending on how you triggered them)

And this currently forbids having additional requests being triggered before or in-between expected ones, but not after them.

Reply:
I have 2 ideas to add a check on the method:

```php
$this->assertHttpClientHasBeenCalled($client, 'symfony.http_client', [
    'https://symfony.com/' => [
        'method' => 'GET',
    ],
    'https://symfony.com/doc/current/index.html' => [
        'method' => 'GET',
    ],
]);
```

or

```php
$this->assertHttpClientHasBeenCalled($client, 'symfony.http_client', [
    'GET' => [
        'https://symfony.com/',
        'https://symfony.com/doc/current/index.html'
    ],
]);
```

I prefer the first, because we could add `body` to add a check on the body for POST/PATCH/PUT methods.

Reply:
I would not index by the URL either, because multiple requests might be done for the same URL with different methods (or different body in the case of requests with a body).

I would suggest passing a list of request info, with both the method and the url inside the request info instead.

Or maybe the assertion should care about a single request instead (and if you want to assert that multiple ones have been done, you would run separate assertions). This way, the various checks could be handled as optional arguments of the method instead.